### DISCUSSION TITLE
GitHub cannot search username in organization that have a lot of members

### DISCUSSION THREAD

**Discussion by ramazansancar** [community/community#60018] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

Currently, I can't search any username (including mine) in organization that have a lot of members.

For instance, in the [EpicGames](https://github.com/EpicGames) organization, it will show you no result if you search based on the username.

![image](https://github.com/community/community/assets/8878055/9c569d3a-f887-4956-bece-3777607b4b2a)

And if you force accessing via url with query param (https://github.com/orgs/EpicGames/people?query=ramazansancar), it will give you an "We couldn’t find any matching members." error/info message .

FYI, This issue actually have been reported https://github.com/isaacs/github/issues/1623#issuecomment-606574431 and https://github.com/isaacs/github/issues/1998 but there is no further bug fix.

### Steps to Reproduce
- You must be logged in.
- Just access the [EpicGames organization's poeple](https://github.com/orgs/EpicGames/people) page.
- Search any member username, and get no response 😔
- Force access via url with query param (https://github.com/orgs/EpicGames/people?query=ramazansancar)
- Get your message "We couldn’t find any matching members."

**Comment by prajwalshettydev** [comment]
Issue still exists on sep-2023. 
![image](https://github.com/community/community/assets/18365548/bce017d8-f7f2-44ba-8518-7a9851c557c0)

**Reply by ramazansancar** [reply]
For the solution, see the relevant comment. @prajwalshettydev

**Comment by ramazansancar** [comment]
Similar problem does not work in the user update area either.

When I try to check with my user ID, no data is returned.

![Postman_C1ZaY2Bxxc](https://github.com/community/community/assets/8878055/56ec3f29-bbf4-4751-95a7-1c9407aea0c1)

![chrome_wu3tvh8PTj](https://github.com/community/community/assets/8878055/f4420a07-6a58-44ad-8008-eb776b39adcb)

**Comment by ekkx** [comment, marked as answer]
If your goal is to publicize in your profile that you're a member of the organization, you can use the rest API officially provided by GitHub 👇

https://docs.github.com/en/rest/orgs/members?apiVersion=2022-11-28#set-public-organization-membership-for-the-authenticated-user

Simply generate a personal access token from the settings tab (be sure to enable the `members:write` permission) and request the desired endpoint with it.

**Reply by ramazansancar** [reply]
Thanks. This solved my problem.

Since there is no authority for `members:write`, I created a short-term token and gave all the permissions. After the process was completed, I deleted the token. If there are people who want to do it, they can follow this path.

HTTP: 204 is success
![image](https://github.com/community/community/assets/8878055/b69127e8-b366-4b61-90f6-3d1d14d24ad9)


![image](https://github.com/community/community/assets/8878055/2bcec420-032a-4867-b130-f20fa04e70bf)

**Reply by iBener** [reply]
Selam @ramazansancar 
Thanks this helped me either.

**Reply by ramazansancar** [reply]
Selam @iBener,
You're welcome, have a nice greening

**Reply by prajwalshettydev** [reply]
worked for me, thank you!!
